ocaml/xenopsd/xc/device.ml

```diff
@@ -1953,89 +1934,26 @@ module PCI = struct
          [] (Array.to_list devs)
       )
 
-  let reset ~xs address =
+  let reset ~xs:_ address =
     let devstr = Xenops_interface.Pci.string_of_address address in
     debug "Device.Pci.reset %s" devstr ;
     do_flr devstr
 
-  let clean_shutdown (task : Xenops_task.task_handle) ~xs (x : device) =
+  let clean_shutdown (_ : Xenops_task.task_handle) ~xs (x : device) =
     debug "Device.Pci.clean_shutdown %s" (string_of_device x) ;
     let devs = enumerate_devs ~xs x in
-    Xenctrl.with_intf (fun xc ->
-        try release devs x.frontend.domid with _ -> ()
-    ) ;
-    ()
+    try release devs x.frontend.domid with _ -> ()
 
   let hard_shutdown (task : Xenops_task.task_handle) ~xs (x : device) =
     debug "Device.Pci.hard_shutdown %s" (string_of_device x) ;
     clean_shutdown task ~xs x
 
-  (* This is the global location where PCI device add/remove status is put. We
-     should aim to use a per-device location to support parallel requests in
-     future *)
-  let device_model_state_path xs be_domid fe_domid =
-    Printf.sprintf "%s/device-model/%d/state"
-      (xs.Xs.getdomainpath be_domid)
-      fe_domid
-
-  let signal_device_model ~xs domid cmd parameter =
-    debug "Device.Pci.signal_device_model domid=%d cmd=%s param=%s" domid cmd
-      parameter ;
-    let be_domid = 0 in
-    (* XXX: assume device model is in domain 0 *)
-    let be_path = xs.Xs.getdomainpath be_domid in
-    (* Currently responses go in this global place. Blank it to prevent
-       request/response/request confusion *)
-    xs.Xs.rm (device_model_state_path xs be_domid domid) ;
-    Xs.transaction xs (fun t ->
-        t.Xst.writev be_path
-          [
-            (Printf.sprintf "device-model/%d/command" domid, cmd)
-          ; (Printf.sprintf "device-model/%d/parameter" domid, parameter)
-          ]
-    )
-
   (* Return a list of PCI devices *)
   let list = read_pcidir
-
-  (* We explicitly add a device frontend in the hotplug case so the device watch
-     code can find the backend and monitor it. *)
-  let ensure_device_frontend_exists ~xs backend_domid frontend_domid =
```

Comment:
This appears to be a remnant from  this old commit (which didn't remove all dead code):
```
commit a185ec6e1bee1c0957465cb5e60287c3cb487506
Author: Koushik Chakravarty <koushik.chakravarty@citrix.com>
Date:   Thu Aug 6 11:11:22 2015 +0000

    CA-168029: Using xl pci for plug/unplug in xenopsd-xc

    This xl pci-attach and pci-detach are working now and this
    patch makes xenopsd-xc to use them.
    Also removing some dead code paths
```